src/weight_windows.cpp

```diff
@@ -56,6 +56,9 @@ void apply_weight_windows(Particle& p)
   // skip dead or no energy
   if (p.E() <= 0 || !p.alive())
     return;
+  // skip particles other than neutrons and photons
+  if (p.type() != ParticleType::neutron && p.type() != ParticleType::photon)
+    return;
```

Comment:
This will be handled in `WeightWindows::get_weight_window` when we compare particle types, so I don't think we need it here as well. It does help us avoid allocating a `WeightWindow` object though. Just noting this here
 for posterity I guess. Let's leave it here for now.

Reply:
Addressed in the latest commit